pmd-java/src/main/java/net/sourceforge/pmd/lang/java/JavaLanguageParser.java

```diff
@@ -18,16 +18,19 @@
  */
 public class JavaLanguageParser extends AbstractJavaParser {
     private final int jdkVersion;
+    private final boolean preview;
 
-    public JavaLanguageParser(int jdkVersion, ParserOptions parserOptions) {
+    public JavaLanguageParser(int jdkVersion, boolean preview, ParserOptions parserOptions) {
```

Comment:
This is technically an API breaking change. I'm not sure it matters much, yet we could also add an overloaded constructor with a default value.